src/overlay006/ov6_02240C9C.c

```diff
@@ -1050,332 +973,290 @@ static u8 ov6_022419EC (const int param0)
-                return (u8)v0;
-            }
-        }
+    if (!param1->isFirstMonEgg && (param1->firstMonAbility == ABILITY_SYNCHRONIZE) && (inline_020564D0(2) == 0)) {
+        u32 v0 = Pokemon_GetValue(param0, MON_DATA_PERSONALITY, NULL);
+        return (u8)(v0 % 25);
     }
 
     return inline_020564D0(25);
 }
 
-static u8 ov6_02241B40 (const UnkStruct_ov6_0224222C * param0, const UnkStruct_ov6_022422D0 * param1)
+static u8 ov6_02241B40 (const UnkStruct_ov6_0224222C * param0, const WildEncounters_FieldParams * param1)
 {
-    u8 v0;
     u8 v1;
-    u8 v2, v3;
-    u8 v4;
+    u8 minLevel, maxLevel;
 
-    if (param0->unk_04 >= param0->unk_06) {
-        v2 = param0->unk_06;
-        v3 = param0->unk_04;
+    if (param0->maxLevel >= param0->minLevel) {
+        minLevel = param0->minLevel;
+        maxLevel = param0->maxLevel;
     } else {
-        v2 = param0->unk_04;
-        v3 = param0->unk_06;
+        minLevel = param0->maxLevel;
+        maxLevel = param0->minLevel;
     }
 
-    v0 = v3 - v2 + 1;
-    v1 = LCRNG_Next() % v0;
+    u8 levelRange = maxLevel - minLevel + 1;
+    v1 = LCRNG_Next() % levelRange;
 
-    if (param1->unk_0D == 0) {
-        if ((param1->unk_0E == 55) || (param1->unk_0E == 72) || (param1->unk_0E == 46)) {
-            if (inline_020564D0(2) == 0) {
-                return v2 + v1;
-            }
-
-            return v3;
+    if (!param1->isFirstMonEgg && ((param1->firstMonAbility == ABILITY_HUSTLE) || (param1->firstMonAbility == ABILITY_VITAL_SPIRIT) || (param1->firstMonAbility == ABILITY_PRESSURE))) {
+        if (inline_020564D0(2) == 0) {
+            return minLevel + v1;
         }
+
+        return maxLevel;
     }
 
-    return v2 + v1;
+    return minLevel + v1;
 }
 
-static void ov6_02241BAC (const u16 param0, const u8 param1, const int param2, const u32 param3, const UnkStruct_ov6_022422D0 * param4, Pokemon * param5, BattleParams * param6)
+static void ov6_02241BAC (const u16 species, const u8 level, const int param2, const u32 param3, const WildEncounters_FieldParams * param4, Pokemon * firstPartyMon, BattleParams * battleParams)
 {
-    BOOL v0;
-    u32 v1;
-    BOOL v2;
-    u8 v3;
-    u8 v4;
-    Pokemon * v5;
+    u8 firstMonGender;
+    u8 firstMonNature;
 
-    v5 = Pokemon_New(11);
-    Pokemon_Init(v5);
+    Pokemon * newEncounter = Pokemon_New(11);
+    Pokemon_Init(newEncounter);
 
-    v2 = 0;
+    BOOL abilityInEffect = FALSE;
 
-    if (param4->unk_0D == 0) {
-        if (param4->unk_0E == 56) {
-            u32 v6;
+    if (param4->isFirstMonEgg == FALSE) {
+        if (param4->firstMonAbility == ABILITY_CUTE_CHARM) {
+            u32 speciesGenderRatio = PokemonPersonalData_GetSpeciesValue(species, MON_DATA_PERSONAL_GENDER);
 
-            v6 = PokemonPersonalData_GetSpeciesValue(param0, 18);
-
-            switch (v6) {
+            switch (speciesGenderRatio) {
+            // TODO use Gender ratio enum
             case 0:
             case 254:
             case 255:
```

Comment:
**question:** is there any reason not to use this now?

Reply:
I had only seen the enum defined in [pokemon.c](https://github.com/pret/pokeplatinum/blob/807da949abdbb63901bf1cf7d919603dd3477b6a/src/pokemon.c#L384). Basically the same constants are defined in `build/consts/pokemon.h`. This should probably be cleaned up.

Reply:
I agree! `pokemon.c` should be using the constants defined by our generation process in the build-tree (`build/consts/pokemon.h`). Go ahead and use the ones from the build-tree, and we can address the redefinition in `pokemon.c` in another PR.